src/main/java/org/terasology/signalling/componentSystem/SignalSystem.java

```diff
@@ -181,6 +189,11 @@ public boolean apply(@Nullable SignalNetworkNode input) {
                 });
     }
 
+    /**
+     * Queries each node's Type to determine if it's a Producer
+     * @param network The network to query for Producers
```

Comment:
In line with @pollend's last comment, modify this to
`Finds all the producers in a given network.`